SetupScript.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+if [ $USER -ne "root"]
+	then
+	echo "Please run as root user"
+	exit 1
+fi
+#base script to fresh install systems onto base image for raspberry pi
+
+#check internet connection
+ping -q -c1 google.com > /dev/null
+
+if [ $? -ne 0 ]
+then
+	echo "No Internet Connection...Aborting"
+	exit 1
+fi
+
+#configure displays
+(cd /opt/ && git clone https://github.com/UCSolarCarTeam/Epsilon-Raspberry.git)
+(cd /opt/Epsilon-Raspberry/primary/ && cp xorg.conf /etc/X11/ && cp config.txt /boot/)
+
+#install rabbit-mq
+echo 'deb http://www.rabbitmq.com/debian/ testing main' | tee /etc/apt/sources.list.d/rabbitmq.list && apt-get update && apt-get install rabbitmq-server
+apt-get install cmake libboost-dev openssl libssl-dev libblkid-dev e2fslibs-dev libboost-all-dev libaudit-dev software-properties-common build-essential mesa-common-dev libgl1-mesa-dev
+
+cd /tmp/
```

Comment:
Use a subshell command

Reply:
Why do I use a subshell command here? Since the entire if-statement happens inside the /tmp directory?

Reply:
This is why you should use a subshell command, because conceptually you think you are running everything in the if else statement inside `/tmp/`, but in fact the state for the rest of the script has been affected, and this is never corrected, so alot more than the if else statement is running in `/tmp/` now.